src/services/FileUploadService.ts

```diff
@@ -67,6 +69,9 @@ export class FileUploadService {
         }
         uploadEntry.checksum(checksum);
         const savedEntry = await this.repo.saveEntry(uploadEntry.build());
+        if (expires) {
+            await this.expires(savedEntry, expires);
```

Comment:
can we do this before the entity is saved on the above line.

if you want to do this in another private method, then you can pass in the `uploadEntry` var into it, it's a builder `IBuilder<FileUploadModel>` so you can simply do 
`uploadEntry.customExpires()` inside the method. 

this way we can avoid saving then querying it again. 

To caculate the original retention, also pass in the `resourcePath` string 

`getTImeLeft` relies on `FileUploadModel` might need a change or a new method to take a path instead and use `Date.now()` instead of `entry.createdAt` as it doesn't exist yet, but the times will be about the same.

It is worth noting that `FileEngine` has a `getFileSize` method.

Reply:
Implemented.